pygenn/genn_groups.py

```diff
@@ -632,6 +632,18 @@ def get_sparse_post_inds(self):
                 raise Exception("only manually initialised connectivity "
                                 "can currently by accessed")
 
+            if self.connectivity_initialiser is not None:
+                # download the connectivity details from device
+                self.pull_connectivity_from_device()
```

Comment:
The rest of the API assumes that the user knows when to call "pull" functions so this shouldn't be done automatically in this one case.

Reply:
so this should be called from the PyNN side?

Reply:
Exactly! That also has the advantage that you only need to make that (potentially expensive) call once if you, for instance, want both the pre and post indices